src/components/msgBusSubscriber.ts

```diff
@@ -0,0 +1,58 @@
+import IUsbTransport from '@huddly/sdk-interfaces/lib/interfaces/IUsbTransport';
+import IMsgBusSubscriber, {
+  MsgBusSubscriberOptions,
+} from '@huddly/sdk-interfaces/lib/interfaces/IMsgBusSubscriber';
+import Logger from '@huddly/sdk-interfaces/lib/statics/Logger';
+
+/**
+ * Class for handling framing subscriptions
+ */
+export default class MsgBusSubscriber implements IMsgBusSubscriber {
```

Comment:
If we want improved typing we could make this class generic, where the provided type is whatever message is passed to the subscriptionHandler
```
MsgBusSubscriber<Detections>
```

Reply:
If I understand you correctly that means that we will have to reinstantiate the subscriber when switching framing modes? Not necessarily against that though. It also gave me an idea for the interface

Reply:
Yes that's a good point. Or you could do a discriminated union type and check for whatever is present